include/validator/context.h

```diff
@@ -71,6 +75,38 @@ class Context {
                                      : ComponentContexts.back().Parent;
   }
 
+  size_t getComponentIndexSize(AST::Component::SortCase SC) const noexcept {
+    auto It = getCurrentContext().ComponentIndexSizes.find(SC);
+    return (It != getCurrentContext().ComponentIndexSizes.end()) ? It->second
+                                                                 : 0;
+  }
+
+  size_t getCoreIndexSize(AST::Component::CoreSort CS) const noexcept {
+    auto It = getCurrentContext().CoreIndexSizes.find(CS);
+    return (It != getCurrentContext().CoreIndexSizes.end()) ? It->second : 0;
+  }
+
+  void incrementComponentIndexSize(AST::Component::SortCase SC) noexcept {
```

Comment:
function usually will get a verb i.e. "increase", but here I think we can use shorten "incComponentIndexSize"